packages/node/src/sdk/init.ts

```diff
@@ -130,31 +127,27 @@ function _init(
   }
 
   if (!isCjs()) {
-    const [nodeMajor, nodeMinor] = process.versions.node.split('.').map(Number);
-
-    // Register hook was added in v20.6.0 and v18.19.0
-    if (nodeMajor >= 22 || (nodeMajor === 20 && nodeMinor >= 6) || (nodeMajor === 18 && nodeMinor >= 19)) {
-      // We need to work around using import.meta.url directly because jest complains about it.
-      const importMetaUrl =
-        typeof __IMPORT_META_URL_REPLACEMENT__ !== 'undefined' ? __IMPORT_META_URL_REPLACEMENT__ : undefined;
-
-      if (!GLOBAL_OBJ._sentryEsmLoaderHookRegistered && importMetaUrl) {
-        try {
-          // @ts-expect-error register is available in these versions
-          moduleModule.register('@opentelemetry/instrumentation/hook.mjs', importMetaUrl);
-          GLOBAL_OBJ._sentryEsmLoaderHookRegistered = true;
-        } catch (error) {
-          logger.warn('Failed to register ESM hook', error);
-        }
-      }
-    } else {
-      consoleSandbox(() => {
-        // eslint-disable-next-line no-console
-        console.warn(
-          '[Sentry] You are using Node.js in ESM mode ("import syntax"). The Sentry Node.js SDK is not compatible with ESM in Node.js versions before 18.19.0 or before 20.6.0. Please either build your application with CommonJS ("require() syntax"), or use version 7.x of the Sentry Node.js SDK.',
-        );
-      });
-    }
+    registerEsmModuleHooks([
+      'http',
+      'node:http',
+      'https',
+      'node:https',
+      'connect',
+      'express',
+      'fastify',
+      'graphql',
+      '@hapi/hapi',
+      'ioredis',
+      'koa',
+      'mongodb',
+      'mongoose',
+      'mysql',
+      'mysql2',
+      'pg',
+      '@nestjs/core',
+      'prisma',
+      ...(options._experiments?.modulesToInstrument || []),
```

Comment:
@mydea I added this so that users can include other modules

Reply:
Another more complex approach might be to catch the file-not-found errors and revert to the default resolve/load.

Reply:
> catch the file-not-found errors and revert to the default resolve/load.

I assume this is pretty expensive right?

Reply:
Oh, it looks like it's not possible to fix this way. The errors are coming from the `load` hook while it's fetching files to collect exports.

Looking through the code, `import-in-the-middle` simply doesn't support re-exporting exports from external modules (ie. `export * from 'some-external-module'`). It doesn't appear to support resolving anything that isn't relative to the current file and it needs significant work to fix this. For example, to fix this for mono-repositories too,  it will need to traverse up though multiple the `node_modules` directories searching for the module 🙁

Reply:
Although having written that... I'm going to try a couple more things...

Reply:
Ok, so it's possible to call `resolve` from within `load` to find a modules that cannot be found relative to the current file. I suspect this is how it can be fixed in `import-in-the-middle` so I'll look into doing a PR there. 

I could create a hacky PR similar the this PR but using what I've learnt above but it doesn't fix all the issues and it's almost too hacky to consider actually publishing with the SDK. Once I've got an `import-in-the-middle` PR up, this might change.

The issue with the `@discord` packages at least is that every package has a `version` export which all get combined via `export * from` and results in:
<img width="586" alt="image" src="https://github.com/getsentry/sentry-javascript/assets/1150298/723c3f16-a82c-43f7-897d-aa21d4df9998">